nupic/torch/modules/sparse_weights.py

```diff
@@ -178,35 +159,28 @@ class SparseWeights2d(SparseWeightsBase):
     """
 
     def __init__(self, module, weight_sparsity=None, sparsity=None):
-        assert isinstance(module, nn.Conv2d)
-        assert 0 < (weight_sparsity or sparsity) < 1
+        assert len(module.weight.shape) == 4, "Should resemble a nn.Conv2d"
+        assert 0 <= (weight_sparsity or sparsity) <= 1
```

Comment:
This assert was placed to prevent the creation of SparseWeight2d layers with 0% or 100% sparsity. Both cases should be considered to be a bug.
See comment on `SparseWeights` class